packages/team-assignment/src/components/ListView/index.tsx

```diff
@@ -80,6 +80,69 @@ interface TeamAssignmentViewProps extends RouteComponentProps<RouteParams> {
   defaultPlanId: string;
 }
 
+/**
+ * Function to fetch orgs against a paginatied endpoint
+ *
+ * @param opensrpBaseURL - OpenSRP API base URL
+ * @param pageNumber - paginated page number
+ * @param pageSize - number of orgs in each page
+ */
+async function fetchOrgs(
+  opensrpBaseURL: string,
+  pageNumber: number,
+  pageSize: number
+): Promise<Organization[]> {
+  // pagination params
+  const paginationParams = {
+    pageNumber,
+    pageSize,
+  };
+  // fetch all organizations
+  const organizationsService = new OpenSRPService(ORGANIZATION_ENDPOINT, opensrpBaseURL);
+  try {
+    const getOrgs = await organizationsService.list(paginationParams);
+    return getOrgs;
+  } catch (_) {
+    sendErrorNotification(lang.ERROR_OCCURRED);
+    return [];
+  }
+}
+
+/**
+ * function to fetch paginated orgs resource recursively
+ *
+ * @param opensrpBaseURL - OpenSRP API base URL
+ * @param pageSize - number of orgs in each page
+ * @returns {Promise<Organization[]>} - an array of all orgs in a paginated endpoint
+ */
+async function fetchTeamsRecursively(
+  opensrpBaseURL: string,
+  pageSize: number
+): Promise<Organization[]> {
+  const serve = new OpenSRPService(`organization/count`, opensrpBaseURL);
+  const teamsCount: number = await serve.list();
+
+  // get the maximum possible page numbers
+  const maxPageNo = Math.ceil(teamsCount / pageSize);
+
+  // compose a promise array to resolve in parallel
+  const promises: Promise<Organization[]>[] = [];
+  for (let pageNumber = 1; pageNumber <= maxPageNo; pageNumber++) {
+    promises.push(fetchOrgs(opensrpBaseURL, pageNumber, pageSize));
+  }
+
+  // fetch orgs recursively according to page numbers
+  return Promise.all(promises)
```

Comment:
```suggestion
  const promises: (() => Promise<Organization[]>)[] = [];
  for (let pageNumber = 1; pageNumber <= maxPageNo; pageNumber++) {
    promises.push(() => fetchOrgs(opensrpBaseURL, pageNumber, pageSize));
  }

  // fetch orgs recursively according to page numbers
  return Promise.all(promises.map((prom) => prom()))
```
these functions are evoked immediately instead of being pushed. you might want to wrap them in functions then unwrap at intended invocation.